controllers/productcontroller.go

```
@@ -0,0 +1,74 @@
+	json.NewEncoder(w).Encode(products)
+}
+
+func UpdateProduct(w http.ResponseWriter, r *http.Request) {
+	productId := mux.Vars(r)["id"]
+	if checkIfProductExists(productId) == false {
+		json.NewEncoder(w).Encode("Product Not Found!")
+		return
+	}
+	var product entities.Product
+	database.Instance.First(&product, productId)
+	json.NewDecoder(r.Body).Decode(&product)
```

golangci-lint (1.64.8)

Error return value of `(*encoding/json.Decoder).Decode` is not checked

(errcheck)